lively.morphic/serialization.js

```diff
@@ -81,8 +82,27 @@ export async function saveWorldToResource (world = MorphicEnv.default().world, t
   } finally { i && i.remove(); }
 }
 
-export function copyMorph (morph) {
-  return deserializeMorph(serializeMorph(morph), { migrations, reinitializeIds: true });
+export function copyMorph (morph, realCopy = false) {
+  if (!realCopy) {
+    return deserializeMorph(serializeMorph(morph), { migrations, reinitializeIds: true });
+  }
+  const cachedComments = morph.comments;
+  morph.comments = [];
+
+  let cachedConnections = [];
+  if (morph.attributeConnections) {
+    cachedConnections = morph.attributeConnections.filter(ac => ac.targetObj.isCommentIndicator);
+    morph.attributeConnections = morph.attributeConnections.filter(ac => !(ac.targetObj.isCommentIndicator || ac.targetObj.isHalo));
+  }
+
+  const serializedMorph = serializeMorph(morph);
+
+  morph.comments = cachedComments;
+  if (morph.attributeConnections) {
+    morph.attributeConnections = morph.attributeConnections.concat(cachedConnections);
+  }
+
+  return deserializeMorph(serializedMorph, { migrations, reinitializeIds: true });
```

Comment:
So in principle that is fine, however one thing that has started to bug me is that the Morph class is waaaay to complex already and I am constantly looking for ways to move code into other modules. In fact it is still one of the main points in my agenda to wipe all the code from lively.morphic that is not essential to rendering stuff and moving it to other modules. This is key to ensuring fast loading times when things are deployed in production. So I personally would prefer to keep logic like this out of Morphic and either solve it by enhancing state monitoring inside the comments system itself or start working with commands instead. _Take a look at how the triggering of layout grids of the halo is handled._

Reply:
but this is a method that is called in the whole world and we added logic but did not write a method ourselves.

how would you put that into another module? 

we needed to do this because when a morph is serialized we want to save it with all of its comments but when a morph is copied the comments should not be copied as well since that would probably not make any sense anymore.

Reply:
For the time being that is fine. Let's push that over to the next refactoring.